File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.java
class Solution {
    public int minOperations(int[] nums, int[] numsDivide) 
    {
        int g=gcdarray(numsDivide);
        //System.out.println(g);
        int c=0;
        Arrays.sort(nums);
        for(int i:nums)
        {
            if (g%i==0)
                return c;
            else
                c++;
        }
        return -1;
        
    }
    public int gcdarray(int a[])
    {
        int result=a[0];
        for(int i:a)
        {
            result=gcd(result,i);
            if (result==1)
                return 1;
        }
        return result;
    }
    public int gcd(int a,int b)
    {
        if (a==0)
            return b;
        return gcd(b%a,a);
    }
}

File: 2344-minimum-deletions-to-make-array-divisible/2344-minimum-deletions-to-make-array-divisible.py
class Solution:
    def minOperations(self, nums: List[int], numsDivide: List[int]) -> int:
        g=numsDivide[0]
        c=0
        for i in numsDivide:
            g=math.gcd(g,i)
        
        nums.sort()
        for i in nums:
            if g%i==0:
                return c;
            else:
                c+=1
        
        return -1
    